docs/001-mana_proposal.md

```diff
@@ -0,0 +1,539 @@
+
+### Mana Toolkit
+In this section, all tools and utility functions for mana will be outlined.
+
+#### Mana Related API endpoints
+ - `/info`: Add own mana in node info response.
+ - `value/allowedManaPledge`: Endpoint that clients can query to determine which nodeIDs are allowed as part of
+   `accessMana` and `consensusMana` fields in a transaction.
+ - `value/sendTransactionByJson`: Add `accessMana`, `consensusMana` and `timestamp` fields to the JSON request.
+
+Add a new `mana` endpoint route:
+ - `/mana`: Return access and consensus mana of the node.
+ - `/mana/all`: Return whole mana map (mana perception of the node).
+ - `/mana/access/nhighest`: Return `n` highest access mana holder `nodeIDs` and their access mana values.
+ - `/mana/consensus/nhighest`: Return `n` highest consensus mana holder `nodeIDs` and their consensus mana values.
+ - `/mana/rank`: Return the top percentile the node belongs to relative to the network. For example, if there are 100 nodes in the
+   network owning mana, and a node is the 13th richest, it means that is part of the top 13% of mana holders, but not the
+   top 12%.
+
+#### Metrics collection
+To study the mana module, following metrics could be gathered:
+
+ - Amount of consensus and access mana present in the network. (amount varies because of `Base Mana 2`).
+ - Amount of mana each node holds.
+ - Number of (and amount of mana) a node was pledged with mana in the last `t` interval.
+ - Mana development of a particular node over time.
+ - Mana rank development of a node over time.
+ - Average pledge amount of a node. (how much mana it receives on average with one pledge)
+ - Mean and median mana holdings of nodes in the network. Shows how even mana distribution is.
+ - Average mana of neighbors.
+
+#### Visualization
+
+Each node calculates mana locally, not only for themselves, but for all nodes in the network that it knows. As a result,
+mana perception of nodes may not be exactly the same at all times (due to network delay, processing capabilities), but
+should converge to the same state. A big question for visualization is which node's viewpoint to base mana visualization on? 
+
+When running a node, operators will be shown the mana perception of their own node, but it also makes sense to
+display the perception of high mana nodes as the global mana perception. First, let's look at how local mana perception
+is visualized for a node:
+
+##### Local Perception
+
+There are two ways to visualize mana in GoShimmer:
+ 1. Node Local Dashboard
+ 2. Grafana Dashboard
+
+While `Local Dashboard` gives flexibility in what and how to visualize, `Grafana Dashboard` is better at storing historic
+data but can only visualize time series. Therefore, both of these ways will be utilized, depending on which suits the best.
+
+`Local Dashboard` visualization:
+ - Pie chart of mana distribution within the network.
```

Comment:
Why a Pie chart? wouldn't be better a plot on x y axis to also compare it to a zipf distribution?

Reply:
Do you mean a histogram? We can do that too.